### PR TITLE
Refs #27736 -- Used decorators for GIS lookup registration.

### PR DESCRIPTION
**Pull request by sir-sigurd**

### CODE REVIEW

File: django/contrib/gis/db/models/fields.py [hunk 1]
```diff
@@ -392,6 +386,11 @@ def get_internal_type(self):
         return "ExtentField"
 
 
+class RasterBandTransform(Transform):
```
Comment (timgraham) [inline django/contrib/gis/db/models/fields.py]: What's the reason for moving this?

File: tests/gis_tests/rasterapp/test_rasterfield.py [hunk 2]
```diff
@@ -130,7 +129,10 @@ def test_all_gis_lookups_with_rasters(self):
         stx_pnt.transform(3086)
 
         # Loop through all the GIS lookups.
-        for name, lookup in gis_lookups.items():
+        for name, lookup in BaseSpatialField.get_lookups().items():
+            if not isinstance(lookup, GISLookup):
+                continue
+
```
Comment (timgraham) [inline tests/gis_tests/rasterapp/test_rasterfield.py]: I wouldn't include a blank line here.

File: django/contrib/gis/db/models/fields.py (hunk 1 above)
Comment (sir-sigurd) [inline django/contrib/gis/db/models/fields.py]: There is no need for this, so I moved it back.

File: tests/gis_tests/rasterapp/test_rasterfield.py (hunk 2 above)
Comment (sir-sigurd) [inline tests/gis_tests/rasterapp/test_rasterfield.py]: fixed

**Pull request merged**